### ISSUE TITLE
Fix commented out tests for skilltrees, and write new tests that fully cover the functionality of the skilltree structs.

### ISSUE THREAD

**Issue by Joshuashou** [uchicago-cs/chiventure#1028]
Assigned to Maya Ghosal and Joshua Shou, for Spring 4

**Comment by Joshuashou** [comment]
Changed skilltree_test and skill_test effects, did these changes on a branch effect_test from effect_v2 so that it doesn't push all at once.

**Comment by Joshuashou** [comment]
Created the pull request to close the issue, there are still small bugfixes we need to make to the code to make it compile properly.

**Comment by Joshuashou** [comment]
Closing this issue, successfully implemented code for new tests based off of effect version2, just waiting on linked pull request to be reviewed and merged into Dev branch.

**Comment by Joshuashou** [comment]
Our pull request successfully implemented code for the new tests that cover a wide range of issues that is linked towards the new implemented effects. We also uncommented out a lot of previous tests that failed to work, and rewrote them so they will work. We created a sampleeffects file that allows for the tests to be ran more smoothly and more organized. There are a number of tests that fail due to code errors in skill_get_all_prerequisites, which will be left as an issue for next year's team to look into. The Pull request will be merged once the new changes regarding commenting becomes resolved.

**Comment by Joshuashou** [comment]
Pull request has now successfully been merged, closing comment has been written on the pull request.

**Comment by firatciftci** [comment]
Issue Score: ✔️+

Comments:
Great work! I only have one thing to comment on:
- The issue did not include a descriptive summary (see https://uchicago-cs.github.io/cmsc22000/project/policies.html#tasks-issues for more details and concrete examples)
